base/redactable_error.go

```diff
@@ -12,14 +12,18 @@ package base
 
 import "fmt"
 
-// A redactable error can be used as a drop-in replacement for a base error (as would have been created via
-// fmt.Errorf), which has the ability to redact any sensitive user data by calling redact() on all if it's
-// stored args.
+// RedactableError is an error that can be used as a drop-in replacement for an error,
+// which has the ability to redact any sensitive data by calling redact() on all of its args.
 type RedactableError struct {
 	fmt  string
 	args []interface{}
 }
 
+var (
+	_ error    = &RedactableError{}
+	_ Redactor = &RedactableError{}
+)
+
 // Create a new redactable error.  Same signature as fmt.Errorf() for easy drop-in replacement.
```

Comment:
This code seems fine, although you can't use `%w` which is always tempting to use around errors, I'd drop a comment. If you wrap this error, I also think the redaction stops working so I worry about this use.

This code is unused though so I wonder if it's worth keeping around?

Reply:
- I agree, this type predated `%w` - we should figure out what to do about that when we have time.
- `RedactableError` is used ... ?

![Screenshot_20230517_142811](https://github.com/couchbase/sync_gateway/assets/1525809/d4cbb8cb-bd45-4347-9f92-a0b025c349d6)